airflow/providers/amazon/aws/hooks/elasticache.py

```diff
@@ -0,0 +1,259 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import re
+from time import sleep
+
+from airflow.exceptions import AirflowException
+from airflow.providers.amazon.aws.hooks.base_aws import AwsBaseHook
+
+
+class ElastiCacheHook(AwsBaseHook):
+    """
+    Interact with AWS ElastiCache
+    """
+
+    # Constants for ElastiCache describe API response keys
+    REPLICATION_GROUPS = 'ReplicationGroups'
+    REPLICATION_GROUP = 'ReplicationGroup'
+    STATUS = 'Status'
+
+    # Constants for ElastiCache delete API response messages
+    RESPONSE_NOT_VALID_STATE = 'not in a valid state to be deleted'
+    RESPONSE_DELETING_STATE = 'has status deleting which is not valid for deletion'
+    RESPONSE_MODIFYING_STATE = 'has status modifying which is not valid for deletion'
+
+    # Constants for ElastiCache replication group status
+    STATUS_CREATING = 'creating'
+    STATUS_AVAILABLE = 'available'
+    STATUS_DELETING = 'deleting'
+    STATUS_CREATE_FAILED = 'create-failed'
+
+    def __init__(
+        self, max_retries=10, exponential_back_off_factor=1, initial_poke_interval=60, *args, **kwargs
+    ):
+        """
+        :param max_retries: Max tries for checking availability of and deleting replication group
+        :param exponential_back_off_factor: Factor for deciding next sleep time
+        :param initial_poke_interval: Initial sleep time in seconds
+        """
+        self.max_retries = max_retries
+        self.exponential_back_off_factor = exponential_back_off_factor
+        self.initial_poke_interval = initial_poke_interval
+
+        super().__init__(client_type='elasticache', *args, **kwargs)
+
+    def create_replication_group(self, config):
+        """
+        Call ElastiCache API for creating a replication group
+        :param config: Python dictionary to use as config for creating replication group
+        :return: Response from ElastiCache create replication group API
+        """
```

Comment:
All of your doc comments need a blank line between prose and params, else it won't render correctly
```suggestion
        """
        Call ElastiCache API for creating a replication group
        
        :param config: Python dictionary to use as config for creating replication group
        :return: Response from ElastiCache create replication group API
        """
```